packages/store/src/patterns/patternMatchers.js

```diff
@@ -380,7 +484,15 @@ const makePatternKit = () => {
         return patt.every((p, i) => checkMatches(specimen[i], p, check, i));
       }
       case 'copyRecord': {
-        if (specimenStyle !== 'copyRecord') {
+        if (isKey(patt)) {
+          // Takes care of patterns which are keys, so the rest of this
+          // logic can assume patterns that are not keys.
+          return check(
+            keyEQ(specimen, patt),
+            X`${specimen} - Must be: ${patt}`,
+          );
+        }
+        if (specimenKind !== 'copyRecord') {
```

Comment:
It still seems to me from a "Reduce redundant code" that this check be once at the top, assuming that `specimenStyle` and `patternStyle` must be `===`

Reply:
Can't. Only applies to three of the cases. Resolving.